### PR TITLE
Deprecate SbDirectoryOpen and SbDirectoryClose APIs

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/302730696

Change-Id: Icdce23666554e47789a9109ea861f5533ced2bf2

### CODE REVIEW

File: starboard/shared/win32/posix_emu/dirent.cc [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      next_entry_offset = directory_info->NextEntryOffset;
+      directory_info_pointer += next_entry_offset;
+    } while (next_entry_offset != 0);
+  }
+
+  if (next_directory_entries.empty()) {
+    return -1;
+  }
+  const char* filename = next_directory_entries.rbegin()->c_str();
+  if (starboard::strlcpy((*dirent)->d_name, filename, kSbFileMaxName)) {
```
[the remark this answers is not in the document]
Comment (y4vor) [reply starboard/shared/win32/posix_emu/dirent.cc:121]: The memory is is allocated in `dirent_buf`. So copy to dirent_buf->d_name.
The `*dirent` is where you should store the result. So `*dirent = dirent_buf`.

File: third_party/musl/src/starboard/dirent.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+    SbDirectory result = SbDirectoryOpen(path, NULL);
+    if (result == kSbDirectoryInvalid){
+        return NULL;
+    }
+    return &result;
+}
+
+int closedir(DIR* dir){
+    SbDirectory sbdir = (SbDirectory)dir;
```
Comment (y4vor) [inline third_party/musl/src/starboard/dirent.c:39]: I don't think we can cast DIR to SbDirectory.
We may need to define a custom DIR type for SB < 16.

Comment (madhurajayaraman) [reply third_party/musl/src/starboard/dirent.c]: The implementation where I cast the pointer wasn't working, so I defined a custom DIR type.

File: net/cert/test_root_certs_unittest.cc [hunk 3]
```diff
@@ -93,21 +94,27 @@ TEST(TestRootCertsTest, OverrideTrust) {
   // certificate should not yet be trusted.
   int flags = 0;
   CertVerifyResult bad_verify_result;
+  SB_LOG(INFO) << "override trust 96";
```
Comment (y4vor) [inline net/cert/test_root_certs_unittest.cc:97]: Remove debug lgos in this file?

Comment (madhurajayaraman) [reply net/cert/test_root_certs_unittest.cc:97]: Gah I thought I'd gotten them all!

**Review by y4vor — APPROVED** [review]

**Review by johnxwork — APPROVED** [review]

**Pull request merged**